drivers/pcie/endpoint/pcie_ep_bcm_iproc.c

```diff
@@ -238,6 +240,118 @@ static int iproc_pcie_raise_irq(struct device *dev,
 	return ret;
 }
 
+#if DT_INST_IRQ_HAS_NAME(0, perst)
+static void iproc_pcie_perst(void *arg)
+{
+	struct device *dev = arg;
+	uint32_t data;
+
+	data = sys_read32(CRMU_MCU_EXTRA_EVENT_STATUS);
+
+	if (data & PCIE0_PERST_INTR) {
```

Comment:
I think you have these checks just for completeness otherwise this handler is dedicated for the particular event and the check is not mandatory.

Reply:
Yes, I think it is safe to remove `CRMU_MCU_EXTRA_EVENT_STATUS` read from both PERST and Inband PERST interrupt handlers. Let me double check.

Reply:
Keeping it for now (for safety). Will remove it later once verified not required.